src/test/java/org/assertj/core/api/BDDAssumptionsTest.java

```diff
@@ -0,0 +1,1529 @@
+
+import static org.assertj.core.api.BDDAssertions.thenCode;
+import static org.assertj.core.api.BDDAssertions.thenThrownBy;
+import static org.assertj.core.api.BDDAssumptions.given;
+import static org.assertj.core.api.BDDAssumptions.givenObject;
+
+import java.io.ByteArrayInputStream​;
+import java.io.File;
+import java.io.InputStream;
+import java.math.BigInteger;
+import java.math.BigDecimal;
+import java.net.URI;
+import java.net.URL;
+import java.nio.file.Path;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.time.OffsetTime;
+import java.time.ZonedDateTime;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalDouble;
+import java.util.OptionalInt;
+import java.util.OptionalLong;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Future;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicIntegerArray;
+import java.util.concurrent.atomic.AtomicIntegerFieldUpdater;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicLongArray;
+import java.util.concurrent.atomic.AtomicLongFieldUpdater;
+import java.util.concurrent.atomic.AtomicMarkableReference;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.atomic.AtomicReferenceArray;
+import java.util.concurrent.atomic.AtomicReferenceFieldUpdater;
+import java.util.concurrent.atomic.AtomicStampedReference;
+import java.util.function.DoublePredicate;
+import java.util.function.IntPredicate;
+import java.util.function.LongPredicate;
+import java.util.function.Predicate;
+import java.util.stream.DoubleStream;
+import java.util.stream.IntStream;
+import java.util.stream.LongStream;
+import java.util.stream.Stream;
+
+import org.junit.AssumptionViolatedException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import static org.mockito.BDDMockito.mock;
+import static org.mockito.BDDMockito.willReturn;
+
+/**
+ * Tests for <code>{@link org.assertj.core.api.BDDAssumptions}</code>.
+ *
+ * @author Gonzalo Müller
+ */
+public class BDDAssumptionsTest {
+  @Nested
+  public class BDDAssumptions_given_boolean_Test {
+    private final boolean actual = true;
+
+    @Test
+    public void should_run_test_when_assumption_passes() {
+      thenCode(() -> given(actual).isTrue())
+        .doesNotThrowAnyException();
+    }
+
+    @Test
+    public void should_ignore_test_when_assumption_fails() {
+      thenThrownBy(() -> given(actual).isFalse())
+        .isInstanceOf(AssumptionViolatedException.class);
```

Comment:
let's extract a static util method `expectAssumptionViolatedException(ThrowingCallable shouldRaiseThrowable)` in `AssertionsUtil` to simply write every `should_ignore_test_when_...` like:
```
expectAssumptionViolatedException(() -> given(actual).isFalse());
```
we could go even further and make `should_ignore_test_when_assumption_fails()` a parameterized test taking a `ThrowingCallable` parameters

Reply:
if we do that then the Nested class are not so useful anymore and should be changed to simple test like `should_run_test_when_boolean_assumption_passes`, `should_run_test_when_boolean_array_assumption_passes`, ...

Reply:
I'm doing partially changes, I do not understood all the idea. Did you mean remove `actual` field, and pass it as argument so nested tested are removed and may use Parameterized tests.
Something like this:
```
    @Test
    public void should_run_test_when_boolean_assumption_passes() {
      thenCode(() -> given(true).isTrue())
        .doesNotThrowAnyException();
    }

    @Test
    public void should_ignore_test_when_boolean_assumption_fails() {
      expectAssumptionViolatedException((true) -> given(actual).isFalse());
    }
```
and then parameterized:
```
Thanks!
    @ParameterizedTest
    // ... parameters
    public void should_run_test_when_assumption_passes(ThrowingCallable mayRaiseError) {
      thenCode(mayRaiseError)
        .doesNotThrowAnyException();
    }

    @ParameterizedTest
    // ... parameters
    public void should_ignore_test_when_assumption_fails(ThrowingCallable shouldRaiseError) {
      expectAssumptionViolatedException(shouldRaiseError);
    }
```

Reply:
For some tests this "may be not" possible,  e.g. `BDDAssumptions_given_Comparable_Test`.